### ISSUE TITLE
Could you provide wheels for python 3.12 ?

### ISSUE THREAD

**Issue by thebaptiste** [ecmwf/ecmwflibs#33]
### Describe the solution you'd like

I would like to use many-linux wheel for python 3.12

### Describe alternatives you've considered

I don't see any alternative...

### Organisation

Meteo-France

**Comment by thebaptiste** [comment]
As I'm building eccodes, Magics and Metview from sources and since there is now a workaround on climetlab to install it without ecmwlibs, I hope I won't need anymore ecmwflibs which has caused me a lot of trouble over the years...

**Comment by ghaarsma** [comment]
I second the request for Python 3.12 Wheels. The ecmwflibs package is the last package for us that prevents us from upgrading to Python 3.12

Kindly asking for Python 3.12 Wheels for Linux, Windows and Mac OS X when you release the next version of ecmwflibs.

**Comment by christegho** [comment]
Wondering if any wheel is available for python3.12

**Comment by eschalkargans** [comment]
Hello, I would also be interested into using this project with Python 3.12

Is there anything preventing the upgrade for now?

Thanks!

**Comment by ghaarsma** [comment]
I recommend upgrading to eccodes 2.39.0 (or later) which should run under 3.9 <= Python <= 3.12 for all the major platforms. `eccodes >= 2.39.0` has the eccodes library build in, so no need anymore for the separate ecmwflibs dependency.

**Comment by ameraner** [comment]
Looks like wheels for 3.12 where published in the meanwhile: https://anaconda.org/conda-forge/ecmwflibs/files

**Comment by thebaptiste** [comment]
...but still not on PyPI : https://pypi.org/project/ecmwflibs/#files

**Comment by ghaarsma** [comment]
I recommend upgrading to [eccodes >=2.39.2](https://pypi.org/project/eccodes/), it has the eccodes library build in, so need for the separate ecmwflibs dependency. Eccodes 2.39.2 (or later) is being maintained and will run with Python 3.13.

**Comment by thebaptiste** [comment]
Yes, but I'm also using climetlab, the latest release of which requires ecmwlibs as a dependency (even if I don't need it because all the required libraries are provided elsewhere)...
I will probably fork ecmwlibs as an empty package...

**Comment by ameraner** [comment]
> I recommend upgrading to [eccodes 2.39.2](https://pypi.org/project/eccodes/), it has the eccodes library build in, so need for the separate ecmwflibs dependency. Eccodes 2.39.2 is being maintained and will run with Python 3.13.

@ghaarsma  Did you maybe mean 2.37.0? That's what I got from the pypi page you linked https://pypi.org/project/eccodes/ .
2.39.2 doesn't seem to be existing...

**Comment by ghaarsma** [comment]
> 
> [@ghaarsma](https://github.com/ghaarsma) Did you maybe mean 2.37.0? That's what I got from the pypi page you linked https://pypi.org/project/eccodes/ . 2.39.2 doesn't seem to be existing...

I linked to the general [eccodes](https://pypi.org/project/eccodes/) page which just shows the latest version (currently 2.40.0), but the packages gets multiple updates per year. The actual [2.39.2](https://pypi.org/project/eccodes/2.39.2/) is available on pypi, but if you use PIP or other packages managers like [Poetry](https://python-poetry.org/) or [UV](https://docs.astral.sh/uv/), it should in general install the latest version.

**Comment by odimko** [comment]
hi @danovaro or other contributors, just like others above, i am curious to know is Python 3.12 support (and publishing to PyPi) is planned at some point. Could you please let us know?

**Comment by lu-44** [comment]
Same +1, please provide wheels for python 3.12 on pypi

**Comment by djhoese** [comment]
I understood @ghaarsma comment [here](https://github.com/ecmwf/ecmwflibs/issues/33#issuecomment-2499631553) to mean no one should use `ecmwflibs` for eccodes anymore and instead should use the `eccodes` on PyPI directly. The last release of that package includes Python 3.12 and 3.13 wheels:

https://pypi.org/project/eccodes/#files

**Comment by djhoese** [comment]
@lu-44 No need to comment multiple times. Read my comment above and please explain your situation if the "eccodes" package does not solve your problem.

**Comment by lu-44** [comment]
> [@lu-44](https://github.com/lu-44) No need to comment multiple times. Read my comment above and please explain your situation if the "eccodes" package does not solve your problem.

Yes sorry for that, I deleted the duplicate. 
Still need ecmwflibs  update as it is a recursive dependency in my project and I don't want/have the ability to fix that.
